File: commands/music.js
const { tokenManager } = require("../tokenManager");
const path = require("path");

let mediaManager = null;

module.exports = {
  run: async ({ username, message }) => {
    try {
      const MUSIC_COST = 10; 
      
      const args = message.split(' ');
      if (args.length < 2) {
        await global.sendChatMessage(`@${username}: Usage: >music YouTube_URL (costs ${MUSIC_COST} tokens)`);
        return;
      }
      
      const musicUrl = args.slice(1).join(' ').trim();
      
      if (!musicUrl.match(/(youtube\.com\/watch\?v=|youtu\.be\/)/)) {
        await global.sendChatMessage(`@${username}: Invalid YouTube URL! Please provide a valid YouTube link.`);
        return;
      }
      
      const hasEnough = await tokenManager.hasTokens(username, MUSIC_COST);
      if (!hasEnough) {
        const userData = await tokenManager.getUser(username);
        await global.sendChatMessage(`@${username}: Not enough tokens! You have ${userData.tokens}, need ${MUSIC_COST}. Buy tokens with kicks.`);
        return;
      }
      
      if (!mediaManager) {
        mediaManager = global.mediaManager;
        if (!mediaManager) {
          await global.sendChatMessage(`@${username}: Media system not available right now.`);
          return;
        }
      }
      
      const newBalance = await tokenManager.spendTokens(username, MUSIC_COST, "Music Request");
      
      await global.sendChatMessage(`@${username}: Processing music request... (${MUSIC_COST} tokens spent, ${newBalance} remaining)`);
      
      try {
        const result = await mediaManager.queueMedia(musicUrl, username, 'music');
        
        if (result.success) {
          await global.sendChatMessage(`@${username}: "${result.title}" added to queue! Position: ${result.position}`);
        } else {
          throw new Error("Failed to queue music");
        }
        
      } catch (error) {
        console.error("Music request failed:", error);
        
        await tokenManager.addTokens(username, MUSIC_COST, "Music refund - request failed");
        
        if (error.message.includes("too long")) {
          await global.sendChatMessage(`@${username}: ${error.message} Tokens refunded.`);
        } else if (error.message.includes("Invalid YouTube URL")) {
          await global.sendChatMessage(`@${username}: Invalid YouTube URL. Tokens refunded.`);
        } else if (error.message.includes("HTTP Error 403") || error.message.includes("403")) {
          await global.sendChatMessage(`@${username}: Music access blocked (403 error). Tokens refunded.`);
        } else {
          await global.sendChatMessage(`@${username}: Music request failed. Tokens refunded.`);
        }
      }
      
    } catch (error) {
      console.error("Error in music command:", error);
      await global.sendChatMessage(`@${username}: Error processing music request`);
    }
  }
};

File: KickBot/Commands/Music.cs
namespace KickBot.Commands;

public class MusicCommand : ICommand
{
    public string Name => "music";
    public string Description => "Queue a music request from YouTube";
    public bool BroadcasterOnly => false;

    public async Task ExecuteAsync(string username, string[] args, CommandContext context)
    {
        if (context.MediaManager == null)
        {
            await context.SendMessage(context.BroadcasterUserId, $"@{username} Media requests are not enabled!");
            return;
        }

        if (args.Length == 0)
        {
            await context.SendMessage(context.BroadcasterUserId, $"@{username} Usage: {context.CommandPrefix}music <youtube url> (Cost: {context.MusicCost} tokens)");
            return;
        }

        var userData = context.UserManager.GetOrCreateUser(username);

        if (userData.Tokens < context.MusicCost)
        {
            await context.SendMessage(context.BroadcasterUserId, $"@{username} Not enough tokens! You need {context.MusicCost} tokens but only have {userData.Tokens}.");
            return;
        }

        var url = args[0];

        if (!url.Contains("youtube.com/watch") && !url.Contains("youtu.be/"))
        {
            await context.SendMessage(context.BroadcasterUserId, $"@{username} Invalid YouTube URL!");
            return;
        }

        try
        {
            var result = await context.MediaManager.QueueMedia(url, username, Media.MediaType.Music);

            if (result.Success)
            {
                context.UserManager.SpendTokens(username, context.MusicCost);
                await context.SendMessage(context.BroadcasterUserId, $"@{username} Music queued! Position: {result.QueuePosition} (Cost: {context.MusicCost} tokens)");
                Console.WriteLine($"[COMMAND] {username} queued music: {url} - Cost: {context.MusicCost} tokens");
            }
            else
            {
                await context.SendMessage(context.BroadcasterUserId, $"@{username} Failed to queue music: {result.Message}");
            }
        }
        catch (Exception ex)
        {
            await context.SendMessage(context.BroadcasterUserId, $"@{username} Failed to queue music. Please try again.");
            Console.WriteLine($"[ERROR] Music queue failed: {ex.Message}");
        }
    }
}